server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SearchDataLoader.java

```diff
@@ -81,105 +88,125 @@ SearchData load(SearchWsRequest request) {
-    try {
-      for (Language language : languages.all()) {
-        profiles.add(getDefaultProfile(dbSession, language.getKey(), profileName));
-      }
-    } finally {
-      dbClient.closeSession(dbSession);
+    addAll(qualityProfiles, findDefaultProfiles(dbSession, languageKeys));
+    return difference(languageKeys, qualityProfiles.keySet());
+  }
+
+  private static <T> Set<T> difference(Set<T> languageKeys, Set<T> set2) {
+    return Sets.newHashSet(Sets.difference(languageKeys, set2));
+  }
+
+  private void addAllFromDto(Map<String, QProfile> qualityProfiles, Collection<QualityProfileDto> list) {
+    for (QualityProfileDto qualityProfileDto : list) {
+      qualityProfiles.put(qualityProfileDto.getLanguage(), QualityProfileDtoToQProfile.INSTANCE.apply(qualityProfileDto));
     }
+  }
 
-    return profiles;
-  }
-
-  private static QProfile profileDtoToQProfile(QualityProfileDto dto) {
-    return new QProfile()
-      .setKey(dto.getKey())
-      .setName(dto.getName())
-      .setLanguage(dto.getLanguage())
-      .setDefault(dto.isDefault())
-      .setRulesUpdatedAt(dto.getRulesUpdatedAt());
-  }
-
-  /**
-   * First try to find a quality profile matching the given name (if provided) and current language
-   * If no profile found, try to find the quality profile set on the project (if provided)
-   * If still no profile found, try to find the default profile of the language
-   * <p/>
-   * Never return null because a default profile should always be set on each language
-   */
-  private QProfile getProfile(DbSession dbSession, String languageKey, @Nullable String projectKey, @Nullable String profileName) {
-    QualityProfileDto profileDto = profileName != null ? profileFactory.getByNameAndLanguage(dbSession, profileName, languageKey) : null;
-    if (profileDto == null && projectKey != null) {
-      profileDto = profileFactory.getByProjectAndLanguage(dbSession, projectKey, languageKey);
+  private void addAll(Map<String, QProfile> qualityProfiles, Collection<QProfile> list) {
+    for (QProfile qProfile : list) {
+      qualityProfiles.put(qProfile.language(), qProfile);
     }
-    profileDto = profileDto != null ? profileDto : profileFactory.getDefault(dbSession, languageKey);
-    checkState(profileDto != null, format("No quality profile can been found on language '%s' for project '%s'", languageKey, projectKey));
+  }
 
-    return profileDtoToQProfile(profileDto);
+  private Set<String> getLanguageKeys() {
+    return from(Arrays.asList(languages.all())).transform(LanguageToKey.INSTANCE).toSet();
   }
 
-  private QProfile getDefaultProfile(DbSession dbSession, String languageKey, @Nullable String profileName) {
-    QualityProfileDto profile = profileName != null ? profileFactory.getByNameAndLanguage(dbSession, profileName, languageKey) : null;
-    if (profile == null) {
-      profile = profileFactory.getDefault(dbSession, languageKey);
+  private List<QProfile> findDefaultProfiles(Set<String> languageKeys) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Private method 'findDefaultProfiles' is never used. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)